### DISCUSSION TITLE
How should I make `mdx.sync` omit Frontmatter in its output using `remark-frontmatter`?

### DISCUSSION THREAD

**Discussion by ajaypillay** [mdx-js/mdx#1590] [Q&A] [answered]
Hi, so I'm trying to figure out how to use [remark-frontmatter](https://github.com/remarkjs/remark-frontmatter). Currently my MDX files are compiled as such:
```
const content = inputFile.getContentsAsString().trim()
const jsx = mdx.sync(content, {
  remarkPlugins: [[remarkFrontmatter, { type: "yaml", marker: "+" }], remarkMath, [remarkPrism, { autolinker: true }], remarkEmbedImages], rehypePlugins: [rehypeKatex, rehypeSlug, rehypeAutolinkHeadings]
})
const jsxComplete = `
import React from 'react'
import { mdx } from '@mdx-js/react'
${jsx}
  `
const output = this.babelCompiler.processOneFileForTarget(inputFile, jsxComplete)

const source = output.data
const sourceMap = JSON.stringify(output.sourceMap)
return { source, sourceMap }
```
However, with the following at the top of my MDX files:
```
+++
title = "Test title"
+++
```
It still gets rendered with the Frontmatter intact. From what I understand, this plugin should be generating a JSON object with the extracted Frontmatter. I'm not seeing this anywhere, and the Frontmatter is still intact in `jsx`.

How exactly should I be doing this? Thanks.

**Comment by ChristianMurphy** [comment, marked as answer]
Start by adjusting the frontmatter marker settings, you have `-` set as the marker
```ts
[remarkFrontmatter, { type: "yaml", marker: "-" }]
```

but in the mdx document, `+` is used

```markdown
+++
title = "Test title"
+++
```

frontmatter will match to the marker type you set it to use, here you set it to parse `-` not `+`.

**Reply by ajaypillay** [reply]
Oh sorry, that's a mistake as I was experimenting around and copied the wrong thing over. It's the same result with no errors thrown when I use `+` as my marker, I tried to use both `-` and `+` as the markers, when I use `-` a `<hr/>` is rendered as it's parsed as a horizontal line.

**Reply by ChristianMurphy** [reply]
What versions of remark-frontmatter and mdx are you using?

MDX 1.x works with remark-frontmatter <=2.x
MDX 2.x works with remark-frontmatter >= 3.0

**Reply by ajaypillay** [reply]
Ah, it was a versioning issue again. `mdx` is at 1.6.22 and I was using `remark-frontmatter` at 3.0.0. I bumped it down to 2.0.0 and now it works, thank you.
